app/src/main/java/com/github/polybooks/core/database/implementation/SaleDatabase.kt

```diff
@@ -175,5 +175,55 @@ class SaleDatabase : SaleDatabase {
     override fun querySales(): SalesQuery {
         return SalesQuery()
     }
+
+    private fun snapshotToSale(snapshot: QueryDocumentSnapshot): Sale {
+        return Sale(
+            snapshot.getString(SaleFields.TITLE.fieldName)!!,
+            snapshot.getLong(SaleFields.SELLER.fieldName)!!.toInt(),
```

Comment:
Do you mean the ID of the user here ? The ID of the Firebase user is composed of characters and integers so the type Long may not be appropriate here... Maybe a String or a reference to an object of type FirebaseUser would be better

Reply:
Seller in Sale is an int, part of the reason I said Sale should be changed